src/deployment/fabricProvisioningHelpers.ts

```diff
@@ -280,80 +288,32 @@ export async function getAzureActionButtons(
                 : ConnectionDialog.addAccount,
         id: "azureSignIn",
         callback: async () => {
-            const account = await azureAccountService.addAccount();
-            logger.verbose(
-                `Added Azure account '${account.displayInfo?.displayName}', ${account.key.id}`,
-            );
+            // Force sign in prompt
+            await VsCodeAzureHelper.signIn(true);
 
             const accountsComponent = state.formComponents[accountFormComponentId];
             if (!accountsComponent) {
                 logger.error("Account component not found");
                 return;
             }
 
-            accountsComponent.options = await getAccounts(azureAccountService, logger);
+            const azureAccounts = await VsCodeAzureHelper.getAccounts();
+            accountsComponent.options = azureAccounts.map((account) => ({
+                displayName: account.label,
+                value: account.id,
+            }));
 
             logger.verbose(
                 `Read ${accountsComponent.options.length} Azure accounts: ${accountsComponent.options.map((a) => a.value).join(", ")}`,
             );
 
-            state.formState.accountId = account.key.id;
-            logger.verbose(`Selecting '${account.key.id}'`);
+            state.formState.accountId = azureAccounts.length > 0 ? azureAccounts[0].id : "";
+            logger.verbose(`Selecting '${azureAccounts[0].id}'`);

```

Comment:
This line will throw if `azureAccounts` is empty.  If there is always expected to be at least one account in this array, then wrap these in an `if` statement that enforces that, and log or throw if it for some reason isn't true.